src/js/angular/rest/graph-data.rest.service.js

```diff
@@ -136,4 +153,32 @@ function GraphDataRestService($http) {
             }
         });
     }
+
+    function resolveGraphs(scope, selectedGraph) {
```

Comment:
I have a couple of considerations here. 
* Services should not deal with UI stuf in either way! Never :)... The problem with this is that services are singletons in angular and we might easily reach to some nasty memmory leaks and degraded performance. Some issues which @teodossidossev is currently dealing with. So, my advice is to remove this `scope` parameter and just pass the resolved values from the method clients e.g. pass `isActiveRepository` and `activeRepository` paramaters instead.
* I don't get the idea of using the `RDF4JRepositoriesRestService` in this `GraphDataRestService`. Why don't you move this method in RDF4JRepositoriesRestService if you use the same endpoint method?
*